platforms/apple/build-xcframework.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+"""
+The script builds OpenCV.framework for OSX.
+This script builds OpenCV into an xcframework compatible the platforms
+of your choice. Just run it and grab a snack; you'll be waiting a while.
+"""
+
+import sys, os, argparse, pathlib, traceback
+from cv_build_utils import execute, print_error, print_header, get_xcode_version, get_cmake_version
+
+def get_framework_build_command_for_platform(platform, destination, framework_name, only_64_bit=False):
+    """
+    Generates the build command that creates a framework supporting the given platform.
+    This command can be handed off to the command line for execution.
+
+    Parameters
+    ----------
+    platform : str
+        The name of the platform you want to build for. Options are
+        "ios", "ios-simulator", "ios-maccatalyst", "macos"
+    destination : str
+        The directory you want to build the framework into.
+    framework_name : str
+        The name of the generated framework.
+    only_64_bit : bool, optional
+        Build only 64-bit archs, by default False
+    """
+    destination = destination.replace(" ", "\\ ")  # Escape spaces in destination path
+    if platform == 'macos':
+        return ["python3", "../osx/build_framework.py", "--archs", "x86_64,arm64", "--framework_name", framework_name, "--build_only_specified_archs", destination]
```

Comment:
This path needs to be relative to the script itself, not the current working directory